internal/test_cases/single_line_output_test_case.go

```diff
@@ -0,0 +1,101 @@
+package test_cases
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/codecrafters-io/shell-tester/internal/shell_executable"
+	"github.com/codecrafters-io/tester-utils/logger"
+)
+
+// SingleLineOutputTestCase verifies a prompt exists, sends a command and matches the output against a string.
+type SingleLineOutputTestCase struct {
+	// The command to execute (the command's output will be matched against ExpectedPattern)
+	Command string
+
+	// ExpectedPattern is the regex that is evaluated against the command's output.
+	// Add \r\n at the end of the pattern if you're expecting a newline.
```

Comment:
Don't think that this ("Add \r\n at the end") needs to be a requirement anymore. If we're expecting a single line, wouldn't it always have a `\r\n` at the end?